File: sdks/php/src/Exceptions/MobiscrollConnectException.php
<?php

declare(strict_types=1);

namespace Mobiscroll\Connect\Exceptions;

use Exception;

class MobiscrollConnectException extends Exception
{
    public function __construct(
        string $message = '',
        private string $code_string = 'MOBISCROLL_ERROR',
        int $code = 0,
        ?Exception $previous = null
    ) {
        parent::__construct($message, $code, $previous);
    }

    public function getCodeString(): string
    {
        return $this->code_string;
    }
}

File: sdks/dotnet/src/Mobiscroll.Connect/Exceptions/MobiscrollConnectException.cs
using System;

namespace Mobiscroll.Connect.Exceptions;

public class MobiscrollConnectException : Exception
{
    public string CodeString { get; }

    public MobiscrollConnectException(string message, string codeString = "MOBISCROLL_ERROR")
        : base(message)
    {
        CodeString = codeString;
    }

    public MobiscrollConnectException(string message, string codeString, Exception innerException)
        : base(message, innerException)
    {
        CodeString = codeString;
    }
}
